overlord/fdestate/backend/reseal.go

```diff
@@ -221,12 +221,11 @@ func doReseal(manager FDEStateManager, method device.SealingMethod, rootdir stri
 		// Note this is the fallback key path for old installations. The file might not exist.
 		primaryKeyFile := filepath.Join(boot.InstallHostFDESaveDir, "aux-key")
 		// All devices should have the same primary key
-		primaryKey, err := secbootGetPrimaryKey(devices, primaryKeyFile)
-		if err != nil {
-			return err
+		primaryKeyGetter := func() ([]byte, error) {
+			return secbootGetPrimaryKey(devices, primaryKeyFile)
```

Comment:
will we have issues related to managing the state lock here?  it seems atm we are releasing it only when we recalculate the profiles, which happens before this?

Reply:
This function is not supposed to do anything with the state. The fallback primary key file is not supposed to change. And the devices are only used to make the name of the key in the keyring. The primary in the keyring is not supposed to change either. So it should be possible to run it concurrently.
But maybe there is something I do not see here.